internal/resources/otelcollector/rbac.go

```diff
@@ -112,6 +112,32 @@ func makeMetricGatewayClusterRole(name types.NamespacedName, kymaInputAllowed bo
 		})
 	}
 
+	if k8sClusterReceiverAllowed {
+		clusterRules := []rbacv1.PolicyRule{{
+			APIGroups: []string{""},
+			Resources: []string{"events", "namespaces/status", "nodes", "nodes/spec", "pods", "pods/status", "services", "replicationcontrollers", "replicationcontrollers/status", "resourcequotas", "namespaces", "pods"},
+			Verbs:     []string{"get", "list", "watch"},
+		}, {
+			APIGroups: []string{"apps"},
+			Resources: []string{"deployments", "daemonsets", "replicasets", "statefulsets"},
+			Verbs:     []string{"get", "list", "watch"},
+		}, {
+			APIGroups: []string{"extensions"},
+			Resources: []string{"deployments", "daemonsets", "replicasets", "statefulsets"},
```

Comment:
Do we need the `statefulsets` in the list of resources for the `extensions` APIGroup? [Here](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/k8sclusterreceiver#rbac) it only listed in the `apps` APIGroup and not in the `extensions` APIGroup